airflow/jobs/local_task_job.py

```diff
@@ -161,3 +166,55 @@ def heartbeat_callback(self, session=None):
                 ti.task.on_success_callback(context)
             self.task_runner.terminate()
             self.terminating = True
+
+    @classmethod
+    def get_zombie_running_tis(cls, limit_dttm, session=None):
+        def batch_get(acc, items):
+            # Redis has no batch functionality for zscore
+            # see https://github.com/antirez/redis/issues/2344
+            lua_script = '''
+                local res = {}
+                while #ARGV > 0 do
+                    res[#res+1] = redis.call('ZSCORE', KEYS[1], table.remove(ARGV, 1))
+                end
+                return res
+            '''
+            from airflow.jobs import LocalTaskJob
+            return acc + cls.redis.register_script(lua_script)(keys=[LocalTaskJob.__name__], args=items)
+
+        TI = airflow.models.TaskInstance
+
+        if conf.getboolean('heartbeat', 'redis_enabled'):
```

Comment:
Get Boolean is a relatively expensive call, we should avoid calling this repeatedly